dask/dataframe/core.py

```diff
@@ -3080,20 +3080,22 @@ def idxmaxmin_chunk(x, fn, axis=0, skipna=True, **kwargs):
     value = getattr(x, minmax)(axis=axis, skipna=skipna)
     n = len(x)
     if isinstance(idx, pd.Series):
-        return pd.DataFrame({'idx': idx, 'value': value, 'n': [n] * len(idx)}).T
-    return pd.DataFrame({'idx': [idx], 'value': [value], 'n': [n]}).T
+        chunk = pd.DataFrame({'idx': idx, 'value': value, 'n': [n] * len(idx)})
+        chunk['idx'] = chunk['idx'].astype(type(idx.iloc[0]))
+    else:
+        chunk = pd.DataFrame({'idx': [idx], 'value': [value], 'n': [n]})
+        chunk['idx'] = chunk['idx'].astype(type(idx))
+    print(chunk)
+    return chunk.T
```

Comment:
At first glance nothing jumps out at me.  

I would step through all of the stages with pdb or print statements as you're doing here.  Perhaps put an `import pdb; pdb.set_trace()` or `print` inside of the `_concat` function to see if it is misbehaving as you suggest?